File: src/main/java/org/quanta/frontend/Frontend.java
package org.quanta.frontend;


import org.antlr.v4.runtime.Lexer;
import org.antlr.v4.runtime.tree.ParseTree;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.CharStreams;
import org.antlr.v4.runtime.CharStream;
import org.quanta.antlr.QuantaLexer;
import org.quanta.antlr.QuantaParser;


public class Frontend {
    public ParseTree parse(String src) {
        CharStream input = CharStreams.fromString(src);
        Lexer lexer = new QuantaLexer(input);
        CommonTokenStream tokens = new CommonTokenStream(lexer);
        QuantaParser parser = new QuantaParser(tokens);
        return parser.program();
    }
}

File: compiler0/frontend.py
from .generated import QuantaParser, QuantaLexer, QuantaVisitor
from antlr4 import *


def parse(src: str) -> QuantaParser.ProgramContext:
    lexer = QuantaLexer(InputStream(src))
    stream = CommonTokenStream(lexer)
    parser = QuantaParser(stream)
    tree = parser.program()
    return tree

def parse_debug(src: str) -> QuantaParser.ProgramContext:
    lexer = QuantaLexer(InputStream(src))
    stream = CommonTokenStream(lexer)
    parser = QuantaParser(stream)
    tree = parser.program()
    print(tree.toStringTree(recog=parser))
    return tree
